K2Bridge/Factories/AggregateFactory.cs

```diff
@@ -0,0 +1,97 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+// See LICENSE file in the project root for full license information.
+
+namespace K2Bridge.Factories
+{
+    using System.Linq;
+    using System.Data;
+    using System.Globalization;
+    using K2Bridge.Models.Response;
+    using Microsoft.Extensions.Logging;
+    using Newtonsoft.Json.Linq;
+    using K2Bridge.Models.Response.Aggregations;
+
+    /// <summary>
+    /// Aggregate Factory.
+    /// </summary>
+    internal static class AggregateFactory
+    {
+        /// <summary>
+        /// Add aggregates to current <see cref="AggregateDictionary"> instance from a given <see cref="DataRow"/>.
+        /// </summary>
+        /// <param name="aggregateDictionary">AggregateDictionary instance.</param>
+        /// <param name="row">The row to be added as aggregate.</param>
+        /// <param name="logger">ILogger object for logging.</param>
+        /// <param name="primaryKey">The primary aggregation key.</param>
+        public static void AddAggregates(this AggregateDictionary aggregateDictionary, DataRow row, ILogger logger, string primaryKey)
+        {
+            var columns = row.Table.Columns;
+
+            foreach (DataColumn column in columns)
+            {
+                if (column.ColumnName == BucketColumnNames.Count || column.ColumnName == primaryKey)
```

Comment:
name == count ?
are we expecting the name to be an int? if its intentional, lets add a comment

Reply:
It means we check if ColumName == "count_" or ColumnName == (a string primary key). There is no more mix between index and real column names. All the values are string.

Reply:
For reference, we were supposing before the primary aggregation column was the first one. Now I am getting the real name of the column (the primary key string).